modules/analysis/src/service.rs

```diff
@@ -0,0 +1,1024 @@
+// use crate::Error;
+use sea_orm::{
+    prelude::ConnectionTrait, ColumnTrait, DbErr, EntityOrSelect, EntityTrait, QueryFilter,
+    QueryOrder, QueryResult, QuerySelect, QueryTrait, Statement,
+};
+use std::collections::HashMap;
+use tracing::instrument;
+use trustify_common::{
+    db::{query::Query, Database, Transactional},
+    model::{Paginated, PaginatedResults},
+};
+
+use crate::model::{
+    AdvisoryDepSummary, AdvisoryGraphSummary, AnalysisStatus, DepNode, GraphMap, PackageNode,
+};
+use crate::Error;
+use itertools::Itertools;
+use petgraph::graph::{Graph, NodeIndex};
+use petgraph::Direction;
+use sea_query::{Condition, Order};
+use std::str::FromStr;
+use trustify_common::db::ConnectionOrTransaction;
+use trustify_common::purl::Purl;
+use trustify_entity::relationship::Relationship;
+use trustify_entity::{sbom, sbom_node};
+
+pub struct AnalysisService {
+    db: Database,
+}
+
+pub fn dep_nodes(
+    graph: &petgraph::Graph<PackageNode, Relationship, petgraph::Directed>,
+    node: NodeIndex,
+) -> Vec<DepNode> {
+    // fast enough to directly instantiate DepNode in this function
+    let mut depnodes: Vec<DepNode> = Vec::new();
```

Comment:
type on the lvalue shouldn't be necessary here, yeah? Inferred since it's used as the `Vec<DepNode>` return.